symengine/sets.cpp

```diff
@@ -804,4 +917,70 @@ RCP<const Set> set_complement(const RCP<const Set> &universe,
     return container->set_complement(universe);
 }
 
+RCP<const Set> conditionset(const RCP<const Basic> &sym,
+                            const RCP<const Boolean> &condition)
+{
+    if (eq(*condition, *boolean(false))) {
+        return emptyset();
+    } else if (eq(*condition, *boolean(true))) {
+        return universalset();
+    }
+    if (is_a<And>(*condition) and is_a<Symbol>(*sym)) {
+        // Simplify if we have a single symbol.
+        const And &aset = down_cast<const And &>(*condition);
+        auto &container = aset.get_container();
+        set_basic present, others;
+        for (auto it = container.begin(); it != container.end(); it++) {
+            if (is_a<Contains>(**it)
+                and eq(*down_cast<const Contains &>(**it).get_expr(), *sym)
+                and is_a<FiniteSet>(
+                        *down_cast<const Contains &>(**it).get_set())) {
+                // iterate through container and check for the condition that
+                // defines the domain of sym.
+                // Simplify if that set is a FiniteSet.
```

Comment:
These types of simplification logic does not belong here. If `And` can be simplified to a `Contains` with a `FiniteSet`, then that should happen in `And`.